index.js

```diff
@@ -174,16 +177,43 @@ Object.defineProperty(File.prototype, 'contents', {
   },
 });
 
+Object.defineProperty(File.prototype, 'cwd', {
+  get: function() {
+    return this._cwd;
+  },
+  set: function(cwd) {
+    if (typeof cwd !== 'string') {
+      throw new Error('cwd should be a string.');
+    }
+    this._cwd = endWithSep(normalize(cwd));
+  },
+});
+
+Object.defineProperty(File.prototype, 'base', {
+  get: function() {
+    return this._base || this._cwd;
+  },
+  set: function(base) {
+    if (base && typeof base !== 'string') {
+      throw new Error('base should be a string.');
+    }
+    this._base = base ? endWithSep(normalize(base)) : null;
+
+    // Make base default to cwd when they are the same
+    if (this._base === this._cwd) {
```

Comment:
But if you normalize after the check, you risk that normalized `base` might now equal `this._cwd`. That's why I check after the normalization, since you need to normalize before the check to be sure they are not the same.